kornia/contrib/distance_transform.py

```diff
@@ -0,0 +1,100 @@
+import math
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from kornia.utils import create_meshgrid
+
+
+def distance_transform(
+    image: torch.Tensor,
+    kernel_size: int = 3,
+    h: float = 0.35
+) -> torch.Tensor:
+    r"""Approximates the Manhattan distance transform of images using cascaded convolution operations.
+
+    The value at each pixel in the output represents the distance to the nearest non-zero pixel in the image image.
+    It uses the method described in :cite:`pham2021dtlayer`.
+    The transformation is applied independently across the channel dimension of the images.
+
+    Args:
+        image: Image with shape :math:`(B,C,H,W)`.
+        kernel_size: size of the convolution kernel.
+        h: value that influence the approximation of the min function.
+
+    Returns:
+        tensor with shape :math:`(B,C,H,W)`.
+
+    Example:
+        >>> tensor = torch.zeros(1, 1, 5, 5)
+        >>> tensor[:,:, 1, 2] = 1
+        >>> dt = kornia.contrib.distance_transform(tensor)
+    """
+    if not isinstance(image, torch.Tensor):
+        raise TypeError(f"image type is not a torch.Tensor. Got {type(image)}")
+
+    if not len(image.shape) == 4:
+        raise ValueError(f"Invalid image shape, we expect BxCxHxW. Got: {image.shape}")
+
+    if kernel_size % 2 == 0:
+        raise ValueError("Kernel size must be an odd number.")
+
+    # n_iters is set such that the DT will be able to propagate from any corner of the image to its far,
+    # diagonally opposite corner
+    n_iters: int = math.ceil(max(image.shape[2], image.shape[3]) / math.floor(kernel_size / 2))
+    grid = create_meshgrid(kernel_size, kernel_size, normalized_coordinates=False,
+                           device=image.device, dtype=image.dtype)
+
+    grid -= math.floor(kernel_size / 2)
+    kernel = torch.hypot(grid[0, :, :, 0], grid[0, :, :, 1])
+    kernel = torch.exp(kernel / -h).unsqueeze(0).unsqueeze(0)
+
+    out = torch.zeros_like(image)
+
+    # It is possible to avoid cloning the image if boundary = image, but this would require modifying the image tensor.
+    boundary = image.clone()
+
+    for i in range(n_iters):
+        cdt = F.conv2d(boundary, kernel, padding='same')

```

Comment:
`padding=='same'` seems that is only supported in pytorch > 1.9
https://discuss.pytorch.org/t/typeerror-conv2d-argument-padding-position-5-must-be-tuple-of-ints-not-str/130513/2

one suggestion is to use `kornia.filters.filter2d` where we handle this already /cc @ducha-aiki 
https://kornia.readthedocs.io/en/latest/filters.html?highlight=filter#kornia.filters.filter2d

I see that for this same reason the numerical test are not consistent

Reply:
I changed it to use filter2d. (Apologies for the wait on responding to these changes, life's been a bit busy!)